lib/charms/mongodb/v1/shards_interface.py

```diff
@@ -893,6 +966,10 @@ def cluster_password_synced(self) -> bool:
             if e.code == 18:  # Unauthorized Error - i.e. password is not in sync
                 return False
             raise
+        except ServerSelectionTimeoutError as e:
+            if e.code == 111:  # Connection refused, - i.e. TLS certs not in sync
```

Comment:
Could you explain what "not in sync" means here?

Reply:
Yes I will add to the comment. This error happens when internal membership is not working across the cluster.

i.e. shard has incorrect internal membership certs and so the config-server cannot communicate with it and vice versa